### PR TITLE
feat: host-based scaffolded function runner

### PR DESCRIPTION
**Pull request by lkingland**
- :gift: host-based scaffolded function runner

/kind enhancement

### CODE REVIEW

File: pkg/functions/runner.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	// Wait for it to become available before returning the metadata.
+	err = waitFor(job, defaultRunTimeout)
+	return
+}
+
+// runFunc returns a function which will run the user's Function based on
+// the jobs runtime.
+func runFunc(ctx context.Context, job *Job) (runFn func() error, err error) {
+	runtime := job.Function.Runtime
+	switch runtime {
```
Comment (lance) [inline pkg/functions/runner.go:80]: Please add `"typescript"` as a case for this switch statement, returning `runnerNotImplemented`.

File: pkg/functions/runner.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	case "java":
+		err = runnerNotImplemented{runtime}
+	case "node":
+		err = runnerNotImplemented{runtime}
+	case "rust":
+		err = runnerNotImplemented{runtime}
+	case "":
+		err = fmt.Errorf("runner requires the function have runtime set")
+	default:
+		err = fmt.Errorf("runner does not recognized the %q runtime", runtime)
```
Comment (lance) [inline pkg/functions/runner.go:94]: Nit.
```suggestion
		err = fmt.Errorf("the %q runtime is not supported", runtime)
```

File: pkg/functions/runner.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	}
+	return
+}
+
+type runnerNotImplemented struct {
+	Runtime string
+}
+
+func (e runnerNotImplemented) Error() string {
+	return fmt.Sprintf("the runtime %q is not supported by the host runner.  Try running containerized.", e.Runtime)
```
Comment (lance) [inline pkg/functions/runner.go:104]:
```suggestion
	return fmt.Sprintf("the runtime %q may only be run containerized.", e.Runtime)
```
Not sure if this is the best wording. I'm just trying to follow convention that errors start with lowercase and do not contain final punctuation. That seems to imply only a single statement.

**Review by lance — COMMENTED** [review]
Looks great! Just a few minor comments.

File: pkg/functions/runner.go (hunk 2 above)
Comment (lance) [reply pkg/functions/runner.go:94]: This wording is not specific enough. My intent is to hide the internal runner resource.

Comment (lkingland) [reply pkg/functions/runner.go:94]: "runner does not recognized"  :wince:  thanks

File: pkg/functions/runner.go (hunk 3 above)
Comment (lkingland) [reply pkg/functions/runner.go:104]: Yes, thank you very much.  Usually when writing these error messages my mind is mostly fully occupied by the logic in development and I tend to not spend enough time ensuring the wording is grammatically correct and informative.

**Comment by lkingland** [comment]
Thanks for the review @lance.  Your improvements to the PR have been made

**Pull request merged**